java/core/src/main/java/com/microsoft/bond/Bonded.java

```diff
@@ -92,6 +92,17 @@
     public abstract void serialize(ProtocolWriter protocolWriter) throws IOException;
 
     /**
+     * Serializes content of this Bonded instance to using the given context and the default Bond type descriptor.
+     * Throws an exception if there is no default type descriptor associated with this Bonded instance.
+     * This method is called when this Bonded is serialized as a field of another object, as opposed to top-level
+     * serializaton. For the top-level serialization, use the public serialize method.
+     *
+     * @param context the serialization context
+     * @throws IOException if an I/O error occurred during serialization
+     */
+    abstract void serialize(BondType.SerializationContext context) throws IOException;
```

Comment:
Should I expect additional test cases in this PR, or are they coming in a different one?